libs/utils/trace.py

```diff
@@ -84,10 +89,21 @@ def __registerTraceEvents(self, events):
             raise ValueError('Events must be a string or a list of strings')
 
 
-    def __parseTrace(self, path, tasks, window):
+    def __parseTrace(self, path, tasks, window, trace_format):
         logging.debug('Loading [sched] events from trace in [%s]...', path)
         logging.debug("Parsing events: %s", self.events)
-        self.ftrace = trappy.FTrace(path, scope="custom", events=self.events, window=window)
+        if trace_format.upper() == 'SYSTRACE' or path.endswith('html'):
+            logging.info('Parsing SysTrace format...')
+            self.ftrace = trappy.SysTrace(path, scope="custom",
+                                          events=self.events,
+                                          window=window)
+            self.trace_format = 'SysTrace'
+        elif trace_format.upper() == 'FTRACE':
+            logging.info('Parsing FTrace format...')
+            self.ftrace = trappy.FTrace(path, scope="custom",
+                                        events=self.events,
+                                        window=window)
```

Comment:
instead of repeating the parameters you could just define which class you are going to use and then call that class outside of the `if` condition.  Simplified:
```
if trace_format.upper() == 'SYSTRACE' or path.endswith('html'):
    logging.info('Parsing SysTrace format...')
    trace_class = trappy.SysTrace
    self.trace_format = 'SysTrace'
elif trace_format.upper() == 'FTRACE':
    logging.info('Parsing FTrace format...')
    self.ftrace = trappy.FTrace
    self.trace_format = 'FTrace'

self.ftrace = trace_class(path, scope="custom", events=self.events, window=window)
```
As I'm going to rebase #81 on top of this, I'll do this in that commit